retriever/lib/provenance.py

```diff
@@ -0,0 +1,247 @@
+import json
+import os
+import pkg_resources
+from collections import OrderedDict
+from datetime import datetime, timezone
+from tempfile import mkdtemp
+from importlib import util
+from shutil import rmtree
+from zipfile import ZipFile
+
+from retriever.engines import choose_engine
+from retriever.lib.datasets import datasets
+from retriever.lib.defaults import HOME_DIR, ENCODING, PROVENANCE_DIR
+from retriever.lib.engine_tools import getmd5
+from retriever.lib.load_json import read_json
+
+
+def package_details():
+    details = {}
+    packages = pkg_resources.working_set
```

Comment:
Have you tested the type of values from `pkg_resources.working_set`.
I think there is something missing. Could you look into this more and try to print out packages.

Reply:
Like what? For me `pkg_resources.working_set` is also giving extra packages in the environment that are not provided using `pip freeze` like `pip`. Also, I couldn't find a better way to find packages without using some external library.
`pip` doesn't provide a python interface.

Reply:
Am not sure why need to split.
how about something like package.version

```python
    details = {}
    packages = pkg_resources.working_set
    for package in iter(pkg_resources.working_set):
        details[package.package_name] = package.version
    return details
```

or

```python
    details = dict([(package.project_name.lower(), package.version)
                    for package in iter(pkg_resources.working_set)])
    return details
```

Reply:
Thanks. This is the better way.